hazelcast/src/main/java/com/hazelcast/config/DurableExecutorConfig.java

```diff
@@ -0,0 +1,196 @@
+     */
+    public static final int DEFAULT_RING_BUFFER_CAPACITY = 100;
+
+    /**
+     * Durability of Executor
+     */
+    public static final int DEFAULT_DURABILITY = 1;
+
+    private String name = "default";
+
+    private int poolSize = DEFAULT_POOL_SIZE;
+
+    private int durability = DEFAULT_DURABILITY;
+
+    private int capacity = DEFAULT_RING_BUFFER_CAPACITY;
+
+    private DurableExecutorConfigReadOnly readOnly;
+
+    public DurableExecutorConfig() {
+    }
+
+    public DurableExecutorConfig(String name) {
+        this.name = name;
+    }
+
+    public DurableExecutorConfig(String name, int poolSize, int durability, int capacity) {
+        this.name = name;
+        this.poolSize = poolSize;
+        this.durability = durability;
+        this.capacity = capacity;
+    }
+
+    public DurableExecutorConfig(DurableExecutorConfig config) {
+        this(config.getName(), config.getPoolSize(), config.getDurability(), config.getCapacity());
+    }
+
+    public DurableExecutorConfigReadOnly getAsReadOnly() {
+        if (readOnly == null) {
+            readOnly = new DurableExecutorConfigReadOnly(this);
+        }
+        return readOnly;
+    }
+
+    /**
+     * Gets the name of the executor task.
+     *
+     * @return The name of the executor task.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Sets the name of the executor task.
+     *
+     * @param name The name of the executor task.
+     * @return This executor config instance.
+     */
+    public DurableExecutorConfig setName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Gets the number of executor threads per member for the executor.
+     *
+     * @return The number of executor threads per member for the executor.
+     */
+    public int getPoolSize() {
+        return poolSize;
+    }
+
+    /**
+     * Sets the number of executor threads per member for the executor.
+     *
+     * @param poolSize The number of executor threads per member for the executor.
+     * @return This executor config instance.
+     */
+    public DurableExecutorConfig setPoolSize(final int poolSize) {
+        if (poolSize <= 0) {
+            throw new IllegalArgumentException("poolSize must be positive");
+        }
+        this.poolSize = poolSize;
+        return this;
+    }
+
+    /**
+     * Gets the durability of the executor
+     *
+     * @return the durability of the executor
+     */
+    public int getDurability() {
+        return durability;
+    }
+
+    /**
+     * Sets the durability of the executor
+     *
+     * @param durability the durability of the executor
+     * @return This executor config instance.
+     */
+    public DurableExecutorConfig setDurability(int durability) {
+        this.durability = durability;
+        return this;
+    }
+
+    /**
+     * Gets the ring buffer capacity of the executor task. 0 means Integer.MAX_VALUE.
```

Comment:
Can you add more info about capacity calculation? User should know it's per partition.. etc.